site/src/components/FavoriteButton.js

```diff
@@ -1,13 +1,42 @@
+import { keyBy, mapValues } from "lodash"
 import React, { useState, useEffect } from "react"
 import classnames from "classnames"
-import { useMutation, useQuery } from "@apollo/react-hooks"
+import { useMutation } from "@apollo/react-hooks"
 import gql from "graphql-tag"
 
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
 import { faHeart as heartFilled } from "@fortawesome/free-solid-svg-icons"
 import { faHeart as heartOutline } from "@fortawesome/free-regular-svg-icons"
 
-import { useAuth0 } from "./Auth0Provider"
+export const GetFavorites = gql`
+  query GetFavorites($loggedIn: Boolean!, $userId: String) {
+    favorites (
+      where: { user_id: { _eq: $userId } }
+    ) @include(if: $loggedIn) {
+      id
+      record_id
+    }
+
+    favorites_count {
```

Comment:
I ended up using a SQL `VIEW` to encapsulate this counting. Couldn't figure out how to do it in one query otherwise. I think this works well for us long-term though if the query ends up taking a while we can cache it in Postgres.